### ISSUE TITLE
NVDA doesn't automatically read the display on the Windows 10 Calculator

### ISSUE THREAD

**Issue by btman16** [nvaccess/nvda#5760]
Hello,

NVDA currently cannot automatically read the new result on the Windows 10 Calculator. I know there is an add-on for this purpose, but having it directly in the core of NVDA would give the following advantages:
1. It will be easier for users coming from a previous version of Windows as the calculator in windows 7/8/8.1 had the display automatically read.
2. Users won't need to go through the extra step of installing an add-on to use an essential part of Windows.

Thanks,

Brandon

**Comment by leonardder** [comment]
@JosephSL: Do you have plans to send in a pr to incorporate this AppModule into core?

**Comment by josephsl** [comment]
Hi, for most situations, live region change event should take care of this. The most annoying side effect as a result is output conflict when speak typed characters is enabled and entering expressions, and if live region change event isn’t careful, it results in duplicate announcements or extraneous text will be announced. I’m investigating this as part of a general upgrade to Windows 10 Calculator support in WinTenApps first before sending it in as a PR, which, due to the executable name, will break any program named “calculator.exe”. Thanks.

**Comment by btman16** [comment]
Hello,

Why is this needed?

I ask because NVDA can already read the display as it changes.

Thanks,

Brandon

 ----- Original Message -----

**Comment by josephsl** [comment]
Hi, imagine a case where someone is typing with speak typed characters enabled. If live region change event handler isn’t careful (implemented in NVDA next snapshots for now), this will result in double-speaking. Most of the code in my add-on deals with this precise case. Thanks.

**Comment by ehollig** [comment]
@josephsl, is this issue resolved? Using the latest master, with speak typed characters enabled, I was able to have calculator announce the result when a formula, such as 2 + 2, was entered without double speaking.

**Comment by josephsl** [comment]
Hi, yes, resolved from the add-on side as it’ll now use live region changed event instead of name change. Thanks.

**Comment by Adriani90** [comment]
@josephsl are you planning to raise a PR to incluse the featur in NVDA's core?

**Comment by josephsl** [comment]
Hi, not at this time, and I’m a bit cautious as Calculator’s user interface and experience keeps changing. Also, the current code is optimized for 2018 releases, and there are people still using older Windows 10 releases, so I’m not sure as to how to go about it. Thanks.

**Comment by btman16** [comment]
Hello,

I'm not sure what the main issue is now. As far as I can tell, NVDA reads Calculator's display fine.

Thanks,

Brandon.

**Comment by josephsl** [comment]
Hi,

The main issue to consider is use of UIA notification event for some things.

As for a general PR to resolve this, I'm queueing it for 2019.2.

Thanks.

**Comment by josephsl** [comment]
Hi,

Update: I'm willing to donate Calculator app module from Windows 10 App Essentials to NVDA Core in the next few moments.

Thanks.

**Comment by josephsl** [comment]
Hi,

For now, I'll close this issue, as the PR I'll be writing addresses a different issue.

Thanks.

**Comment by josephsl** [comment]
Hi,

2022 update: as I mentioned in 2019, I'll close this issue for now. Thanks.
